File: Alz-care-APP/alz-care-api/src/classes/img_classification.py
class Prediction:
    def __init__(self, result, values):
        self.result: str = result
        self.score: list[float] = values


class ImgClassification:

    def __init__(self, plane, prediction, img=None):
        # Validate data
        self._validate(prediction)
        self.plane: str = plane
        self.prediction = prediction
        self.img = img

    @staticmethod
    def _validate(prediction):
        if not (isinstance(prediction, Prediction)):
            raise TypeError()

File: Alz-care-APP/alz-care-front/src/models/ImgClassification.ts
export enum BrainPlanes {
    AXIAL = 'AXIAL',
    CORONAL = 'CORONAL',
    SAGITTAL = 'SAGITTAL'
}

export enum AlzheimerGroupsIndex {
   AD = 0,
   CN = 1,
   MCI = 2,
}

export enum AlzheimerGroupsCode {
   CN = 'Cognitivamente normal',
   MCI = 'Deterioro cognitivo leve',
   AD = 'Alzheimer',
}

export class Prediction {
  /**
   * Alzheimer group result
   */
  groupCode: AlzheimerGroupsCode;

  /**
   * Alzheimer groups score
   */
  score: number[];

  constructor(groupCode: AlzheimerGroupsCode, score: []) {
    this.groupCode = groupCode;
    this.score = score;
  }
}

export class ImgClassification {
  /**
   * Brain plane
   */
  plane: BrainPlanes;
  /**
   * Image
   */
  img: string;
  /**
   * Image prediction
   */
  prediction: Prediction;

  constructor(plane: BrainPlanes, img: string, prediction: Prediction) {
    this.plane = plane;
    this.img = img;
    this.prediction = prediction;
  }
}
